### PR TITLE
feat(header/core): Put eds in `eds.Store` in listener

### LINKED ISSUE #1423: header/core: Put Blocks into the `eds.Store` in `Listener`
## Context

Right now, the `Listener` stores the block data into `ipfs.Blockservice` during `ExtendedHeader` and EDS construction. As per #1099, we are now migrating to new storage for blocks - `eds.Store` and `Listener` should store the EDS/Block there.

This will likely require changes to the `header.MakeExtendedHeader` and likely simplify it by removing dependency on `ipfs.Blockservice`

### PR DESCRIPTION
**Pull request by renaynay**
Something to note about this PR - the level of abstraction introduced by `header.StoreFn` is unfortunately necessary due to cyclical dependencies between `share/eds` and `share/eds/byzantine` where both import the header pkg.

Resolves #1423

### CODE REVIEW

File: das/daser_test.go [hunk 1]
```diff
@@ -5,8 +5,8 @@ import (
 	"testing"
 	"time"
 
-	"github.com/celestiaorg/celestia-node/share/availability/full"
-	"github.com/celestiaorg/celestia-node/share/availability/light"
+	"github.com/celestiaorg/rsmt2d"
+
 	availability_test "github.com/celestiaorg/celestia-node/share/availability/test"
```
Comment (distractedm1nd) [inline das/daser_test.go:10]: this import should be grouped at bottom (but ik wasnt introduced in this pr)

File: header/core/listener.go [hunk 2]
```diff
@@ -20,24 +19,26 @@ import (
...
 type Listener struct {
-	bcast     header.Broadcaster
-	fetcher   *core.BlockFetcher
-	bServ     blockservice.BlockService
+	bcast   header.Broadcaster
+	fetcher *core.BlockFetcher
+
 	construct header.ConstructFn
-	cancel    context.CancelFunc
+	storeFn   header.StoreFn
```
[the remark this answers is not in the document]
Comment (distractedm1nd) [reply header/core/listener.go:26]: also please add godocs to the fields since these callbacks can be confusing

**Review by distractedm1nd — CHANGES_REQUESTED** [review]
first pass

File: header/header.go [hunk 3]
```diff
@@ -76,6 +80,28 @@ func MakeExtendedHeader(
...
+	if err != nil {
+		return da.DataAvailabilityHeader{}, nil, err
+	}
+	size := utils.SquareSize(len(shares))
+	// flatten shares // TODO @renaynay: I wish appshares.Split returned shares as [][]byte instead of the alias.
+	flattened := make([][]byte, len(shares))
+	for i, share := range shares {
+		copy(flattened[i], share)
+	}
+	eds, err := da.ExtendShares(size, flattened)
```
[the remark this answers is not in the document]
Comment (renaynay) [reply header/header.go:96]: bless

**Comment by renaynay** [comment]
FYI - converted to draft as we should wait on #1304 and then move core listener stuff out of header pkg into core pkg.

**Comment by renaynay** [comment]
Superseded by #1632

**Pull request closed without merging**